packages/inter-protocol/test/swingsetTests/psmUpgrade/bootstrap-psm-upgrade.js

```diff
@@ -0,0 +1,313 @@
+
+const scale6 = x => BigInt(Math.round(x * 1_000_000));
+
+const firstGive = anchor.make(scale6(21));
+const secondGive = anchor.make(scale6(3));
+
+/** @typedef {import('../../../src/psm/psm.js').prepare} PsmSF */
+
+export const buildRootObject = async () => {
+  const storageKit = makeFakeStorageKit('psmUpgradeTest');
+  const timer = buildManualTimer();
+  const marshaller = makeBoard().getReadonlyMarshaller();
+
+  const { promise: committeeCreator, ...ccPK } = makePromiseKit();
+
+  /** @type {ZoeService} */
+  let zoeService;
+
+  /** @type {FeeMintAccess} */
+  let feeMintAccess;
+
+  let minted;
+
+  /** @type {PsmPublicFacet} */
+  let psmPublicFacet;
+
+  /** @type {Subscriber<import('../../../src/psm/psm.js').MetricsNotification>} */
+  let metrics;
+  /** @type {UpdateRecord<import('../../../src/psm/psm.js').MetricsNotification>} */
+  let metricsRecord;
+
+  /** @type {VatAdminSvc} */
+  let vatAdmin;
+
+  let initialPoserInvitation;
+  let poserInvitationAmount;
+
+  // for startInstance
+  /**
+   * @type {{
+   * committee?: Installation<import('@agoric/governance/src/committee').start>,
+   * psmV1?: Installation<PsmSF>,
+   * puppetContractGovernor?: Installation<import('@agoric/governance/tools/puppetContractGovernor').start>,
+   * }}
+   */
+  const installations = {};
+
+  /** @type {import('@agoric/governance/tools/puppetContractGovernor').PuppetContractGovernorKit<PsmSF>} */
+  let governorFacets;
+
+  /** @type {Omit<import('@agoric/zoe/src/zoeService/utils.js').StartParams<PsmSF>['terms'], 'issuers' | 'brands'>} */
+  const psmTerms = {
+    anchorBrand: anchor.brand,
+
+    electionManager: Far('mock instance handle'),
+    governedParams: {
+      // @ts-expect-error missing value, to be filled in
+      [CONTRACT_ELECTORATE]: {
+        type: ParamTypes.INVITATION,
+      },
+    },
+  };
+  const staticPrivateArgs = {
+    storageNode: storageKit.rootNode,
+    marshaller,
+  };
+
+  /**
+   * @param {Amount<'nat'>} giveAnchor
+   * @param {Amount<'nat'>} [wantMinted]
+   */
+  const swapAnchorForMintedSeat = async (giveAnchor, wantMinted) => {
+    const seat = E(zoeService).offer(
+      E(psmPublicFacet).makeWantMintedInvitation(),
+      harden({
+        give: { In: giveAnchor },
+        ...(wantMinted ? { want: { Out: wantMinted } } : {}),
+      }),
+      harden({ In: anchor.mint.mintPayment(giveAnchor) }),
+    );
+    return seat;
+  };
+
+  return Far('root', {
+    /**
+     *
+     * @param {{
+     * vatAdmin: ReturnType<import('@agoric/swingset-vat/src/vats/vat-admin/vat-vat-admin')['buildRootObject']>,
+     * zoe: ReturnType<import('@agoric/vats/src/vat-zoe')['buildRootObject']>,
+     * }} vats
+     * @param {*} devices
+     */
+    bootstrap: async (vats, devices) => {
+      vatAdmin = await E(vats.vatAdmin).createVatAdminService(devices.vatAdmin);
+      ({ feeMintAccess, zoeService } = await E(vats.zoe).buildZoe(
+        vatAdmin,
+        undefined,
+        'zcf',
+      ));
```

Comment:
Using promises would make clear that it's a one-time assignment:

```suggestion
      const { feeMintAccess, zoeService } = await E(vats.zoe).buildZoe(
        vatAdmin,
        undefined,
        'zcf',
      );
      bootPK.resolve({ feeMintAccess, zoeService });
```